### PR TITLE
Backport #13568

### PR DESCRIPTION
**Pull request by gsmiller**

### CODE REVIEW

**Comment by stefanvodita** [comment]
I'm surprised at how large the [spotless commit](https://github.com/apache/lucene/pull/13645/commits/fb2c0c50705a6a62bd93d4a270ad113a2e82283e) is. I assume it was just `gradlew tidy` and most code should have already been well formatted.

**Comment by gsmiller** [comment]
@stefanvodita 
> I'm surprised at how large the [spotless commit](https://github.com/apache/lucene/pull/13645/commits/fb2c0c50705a6a62bd93d4a270ad113a2e82283e) is. I assume it was just gradlew tidy and most code should have already been well formatted.

I think what you're seeing is mainly where I had to manually resolve conflicts in the drill-sideways stuff and I was doing it in an editor that kind of botched whitespace, etc. There's a bit indentation change because of bulkScorer moving back out of scoreSupplier among some other stuff...

File: lucene/facet/src/java/org/apache/lucene/facet/DrillSideways.java [hunk 1]
```diff
@@ -240,46 +242,43 @@ public DrillSidewaysResult search(DrillDownQuery query, Collector hitCollector)
...
     for (int dim = 0; dim < numDims; dim++) {
-      List<FacetsCollector> facetsCollectorsForDim = new ArrayList<>(numSlices);
-
-      for (int slice = 0; slice < numSlices; slice++) {
-        facetsCollectorsForDim.add(dsq.managedDrillSidewaysCollectors.get(slice)[dim]);
-      }
-
       drillSidewaysCollectors[dim] =
-          drillSidewaysFacetsCollectorManagers[dim].reduce(facetsCollectorsForDim);
+          (FacetsCollector) drillSidewaysCollectorOwners.get(dim).getResult();
```
Comment (epotyom) [inline lucene/facet/src/java/org/apache/lucene/facet/DrillSideways.java:279]: [nit] I think we can avoid casting here if we add types to `drillSidewaysCollectorOwners`, but given that this method is to be deprecated, I think it's fine to keep casting.

```
final List<CollectorOwner<FacetsCollector, FacetsCollector>> drillSidewaysCollectorOwners = ...
```

**Review by epotyom — APPROVED** [review]
Looks good to me, thanks Greg!

**Pull request merged**